### DISCUSSION TITLE
subtitler - the complete subtitle transcriber and translator

### DISCUSSION THREAD

**Discussion by anupamkumar** [openai/whisper#2248] [Show and tell]
Subtitler is a utility that can be used as GUI or CLI depending on your preference. It uses open-AI's Whisper model to generate a transcript of a video. It additionally can also translate the generated transcript to other languages. All the transcriptions and translations are stored as .srt files.

The most common usecase for using this utility is to auto-generate subtitles for your TVShows, Movies, Home-Videos or auto-generate lyrics for your Music. And not only just auto-generate subtitles and lyrics, but auto-convert them to the language of your preference.

This is most useful when used in conjunction with a Media-Library/Server like Kodi, Emby, Plex or Jellyfin wherein you may have a private library of multimedia and want to enrich it by generating subtitles.

This is a decent alternative to open-subtitles where, subtitles are crowdsourced and can be used to suppliment subtitles not found on such sites.

check it out here - https://github.com/anupamkumar/subtitler

**Comment by fridayflare** [comment]
I'm a YouTuber and I've found the perfect hack for creating English subtitles!

Here's my method: I take my YouTube SBV file (in my local language) and run it through https://subtitletranslatorai.com. It translates it to English, but the grammar is usually a bit rough.

So, I feed that file to an AI (like Gemini or ChatGPT) and give it a simple command: 'Keep the timestamps the same, but fix all the grammar and mistakes.' I also give it a little context on the video, and... it's done!

Perfect subtitles, every single time!
